src/system.rs

```diff
@@ -68,19 +165,108 @@ pub trait System<'a> {
         RunningTime::Average
     }
 
+    /// Return the accessor from the [`SystemData`].
+    fn accessor<'b>(&'b self) -> AccessorCow<'a, 'b, Self> {
+        AccessorCow::Owned(
+            <<Self as System<'a>>::SystemData as SystemData<'a>>::Accessor::try_new()
+                .expect("Missing implementation for `accessor`"),
+        )
+    }
+
     /// Sets up the `Resources` using `Self::SystemData::setup`.
     fn setup(&mut self, res: &mut Resources) {
-        <Self::SystemData as SystemData>::setup(res)
+        <Self::SystemData as SystemData>::setup(&self.accessor(), res)
     }
 }
 
-/// A struct implementing
-/// system data indicates that it
-/// bundles some resources which are
-/// required for the execution.
+/// A static system data that can specify its dependencies at statically (at compile-time).
+/// Most system data is a `StaticSystemData`, the `SystemData` type is only needed for very special
+/// setups.
+pub trait StaticSystemData<'a> {
```

Comment:
Note if you're reading this PR: this is basically the old `SystemData` trait, you can still use it and the new trait will be implemented for you.